### PR TITLE
Add a cookbook entry for a sonoff s2X smart switch. 

### PR DESCRIPTION
**Pull request by rnauber**
## Description:
Just a small example of a firmware for the sonoff s2x series.

## Checklist:
  - [x] The documentation change has been tested and compiles correctly.
  - [X] Branch: `next` is for changes and new documentation that will go public with the next esphomelib release. Fixes, changes and adjustments for the current release should be created against `current`.

### CODE REVIEW

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+      board_flash_mode: dout
+
+
+    wifi:
+      networks:
+        - ssid: !secret wifi_ssid
+          password: !secret wifi_password
+        - ssid: !secret wifi_ssid2
+          password: !secret wifi_password2
+    # power_save_mode: none
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Remove this line

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+      ap:
+        ssid: "$devicename AP"
+        password: !secret  ap_password 
+        channel: 10
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Also remove channel. The example configs here should be as simple as possible (possibly even remove `ap:`)

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+      - platform: monochromatic
+        name: "Green LED"
+        output: green_led
+        id: green_led_light
+        internal: True
+
+
+
+3. Home Assistant
+*******************
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Remove the two * at the end here (heading length must match characters below)

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+An example for an automation that toggles a switch when another switch is double clicked:
+
+.. code-block:: yaml
+    - alias: on double click
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Insert an empty line after `.. code-block:: yaml`. Otherwise it will not show correctly.

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+      trigger:
+      - entity_id: binary_sensor.sonoff1_double_click
+        platform: state
+        from: 'off'
+        to: 'on'
+      condition: []
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Remove this line

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+      ap:
+        ssid: "$devicename AP"
+        password: !secret  ap_password 
+        channel: 10
+
+    mqtt:
+      id: mqtt_client
+      broker: !secret mqtt_broker
+      username: !secret mqtt_user 
+      password: !secret mqtt_password
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: I would prefer advertising `api:` in the documentation. Can we change this example to use `api:` instead of MQTT?

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+        id: long_click
+        lambda: |-
+          return false;   
+
+
+    sensor:
+      - platform: wifi_signal
+        name: "$devicename WiFi Signal"
+        update_interval: 60s
+    #    expire_after: 0
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Remove this

File: esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+            - delay: 0.5s
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - lambda: |-  
+                id(double_click).publish_state(false);
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst]: Please add a comment (or two) as for what this block does. It's quite hard to parse what these `then` blocks are supposed to do.

**Comment by rnauber** [comment]
Thanks for the feedback @OttoWinter, I updated this PR.
